dataclients/kubernetes/kube.go

```diff
@@ -498,9 +502,21 @@ func (c *Client) LoadAll() ([]*eskip.Route, error) {
 	c.current = mapRoutes(r)
 	log.Debugf("all routes loaded and mapped")
 
+	go c.registerSigtermHandler()
+
 	return r, nil
 }
 
+func (c *Client) registerSigtermHandler() {
+	log.Info("register sigterm handler")
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM)
+	// Block until a signal is received.
+	s := <-sigs
+	log.Infof("shutdown, caused by %s, set healthCheck to be unhealty", s)
+	c.healthy = false
```

Comment:
writing here causes a race condition because healthy is accessed from multiple goroutines. Since neither is not a performance critical path, a simple Mutex would be enough here.